packages/teleterm/src/ui/ClusterConnect/ClusterLogin/useClusterLogin.ts

```diff
@@ -75,7 +75,15 @@ export default function useClusterLogin(props: Props) {
   };
 
   useEffect(() => {
-    init();
+    init().then(resp => {
+      const [cfg, err] = resp;
+      if (!err && cfg.preferredMfa === 'u2f') {
+        const statusText = `the U2F API for hardware keys is deprecated, \
+        please notify your system administrator to update cluster \
+        settings to use WebAuthn as the second factor protocol.`;
+        setInitAttempt({ status: 'error', statusText });
+      }
+    });
```

Comment:
Tested this myself on a cluster in our Cloud. u2f handling has to be moved a bit deeper in the call stack, otherwise `init` finishes with success and `ClusterLogin` tries to render `LoginForm` which throws an error because it tries to read `.value` from undefined somewhere.

I made a quick patch to see if changing it to this would work and it does.

<details>
<summary>Patch to useClusterLogin.ts</summary>

```diff
diff --git a/packages/teleterm/src/ui/ClusterConnect/ClusterLogin/useClusterLogin.ts b/packages/teleterm/src/ui/ClusterConnect/ClusterLogin/useClusterLogin.ts
index 03c989be..3ec35641 100644
--- a/packages/teleterm/src/ui/ClusterConnect/ClusterLogin/useClusterLogin.ts
+++ b/packages/teleterm/src/ui/ClusterConnect/ClusterLogin/useClusterLogin.ts
@@ -28,8 +28,16 @@ export default function useClusterLogin(props: Props) {
   const [shouldPromptSsoStatus, promptSsoStatus] = useState(false);
   const [shouldPromptHardwareKey, promptHardwareKey] = useState(false);
 
-  const [initAttempt, init, setInitAttempt] = useAsync(() => {
-    return clustersService.getAuthSettings(clusterUri);
+  const [initAttempt, init] = useAsync(async () => {
+    const authSettings = await clustersService.getAuthSettings(clusterUri);
+
+    if (authSettings.preferredMfa === 'u2f') {
+      throw new Error(`the U2F API for hardware keys is deprecated, \
+        please notify your system administrator to update cluster \
+        settings to use WebAuthn as the second factor protocol.`);
+    }
+
+    return authSettings;
   });
 
   const [loginAttempt, login] = useAsync((opts: types.LoginParams) => {
@@ -75,15 +83,7 @@ export default function useClusterLogin(props: Props) {
   };
 
   useEffect(() => {
-    init().then(resp => {
-      const [cfg, err] = resp;
-      if (!err && cfg.preferredMfa === 'u2f') {
-        const statusText = `the U2F API for hardware keys is deprecated, \
-        please notify your system administrator to update cluster \
-        settings to use WebAuthn as the second factor protocol.`;
-        setInitAttempt({ status: 'error', statusText });
-      }
-    });
+    init();
   }, []);
 
   useEffect(() => {
```
</details>